### PR TITLE
Dev 3520 new go exercise loafofbread

### PR DESCRIPTION
**Pull request by tiagocollot**

### CODE REVIEW

File: solutions/loafofbread.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+func LoafOfBread(str string) string {
+	if str == "" {
+		return "\n"
+	}
+	if len(str) < 5 {
+		return "Invalid Output\n"
+	}
+	s := strings.ReplaceAll(str, " ", "")
+	var _str strings.Builder
```
Comment (eslopfer) [inline solutions/loafofbread.go:13]: This doesn't mean it's wrong, I am just wondering, is there any reason for the leading underscore?

Comment (tiagocollot) [reply solutions/loafofbread.go:13]: A single leading underscore in front of a variable, a function, or a method name means that these objects are used internally, eventually used by a package or variable defined in another folder.

Comment (eslopfer) [reply solutions/loafofbread.go:13]: :+1:

File: solutions/loafofbread.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		if j == 5 {
+			_str.WriteRune(rune(' '))
+			j = 0
+		} else {
+			_str.WriteRune(rune(s[i]))
+			j++
+		}
+	}
+	_str.WriteRune(rune('\n'))
+	return (_str.String())
```
Comment (eslopfer) [inline solutions/loafofbread.go:25]: The solution for "This is a loaf of bread" doesn't match the description of the exercise.

**Review by eslopfer — CHANGES_REQUESTED** [review]

File: tests/loafofbread_test/main.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	student "student"
+
+	"github.com/01-edu/go-tests/lib/challenge"
+	"github.com/01-edu/go-tests/solutions"
+)
+
+func main() {
+	table := []string{"Loaf of bread", "amazing loaf of bread", "bread crumbles", "", "bread slice", "bread"}
```
Comment (eslopfer) [inline tests/loafofbread_test/main.go:11]: Add `"This is a loaf of bread"` to the test cases.
Also, this edge case triggered the thought of more edge cases that are not being tested and could make the exercise more challenging/interesting:
- space at the end of the sentence and it falls between words, when the last word reached 5
- space at the end of the sentence before the last word reaches 5
- space at the beginning of the sentence 
- double spaces
- several different amount of spaces between words
- sentences with only spaces

The solution should work for these as well.

**Review by eslopfer — CHANGES_REQUESTED** [review]

File: solutions/loafofbread.go (hunk 2 above)
Comment (tiagocollot) [reply solutions/loafofbread.go:33]: Fixed.

File: tests/loafofbread_test/main.go (hunk 3 above)
Comment (tiagocollot) [reply tests/loafofbread_test/main.go]: This is a great suggestion! It is done

File: solutions/loafofbread.go (hunk 2 above)
Comment (eslopfer) [reply solutions/loafofbread.go:33]: Thanks

File: tests/loafofbread_test/main.go (hunk 3 above)
Comment (eslopfer) [reply tests/loafofbread_test/main.go]: Thanks
